scripts/keyless/anyEvmSetup.ts

```diff
@@ -0,0 +1,136 @@
+  const [signer] = await hardhatEthers.getSigners();
+  console.log("signer: ", signer.address);
+  const commonFactory = "0x4e59b44847b379578588920cA78FbF26c0B4956C";
+
+  let factoryCode = await hardhatEthers.provider.getCode(commonFactory);
+  // deploy community factory if not already deployed
+  if (factoryCode == "0x") {
+    console.log("zero code");
+    // send balance
+    let refundTx = {
+      to: "3fab184622dc19b6109349b94811493bf2a45362",
+      value: hardhatEthers.utils.parseEther("0.01"),
+    };
+    await signer.sendTransaction(refundTx);
+
+    // deploy
+    try {
+      const tx = await hardhatEthers.provider.sendTransaction(
+        "0xf8a58085174876e800830186a08080b853604580600e600039806000f350fe7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffe03601600081602082378035828234f58015156039578182fd5b8082525050506014600cf31ba02222222222222222222222222222222222222222222222222222222222222222a02222222222222222222222222222222222222222222222222222222222222222",
+      );
+    } catch (err) {
+      console.log("Couldn't deploy factory");
+      console.log(err);
+      process.exit(1);
+    }
+
+    // check
+    factoryCode = await hardhatEthers.provider.getCode(commonFactory);
+    console.log("deployed code: ", factoryCode);
+  }
+
+  // ExtensionRegistry deployment here
+  const bytecodeHash = hardhatEthers.utils.id(ExtensionRegistry.bytecode.object);
+  const salt = `tw.${bytecodeHash}`;
+  const saltHash = hardhatEthers.utils.id(salt);
+
+  const args = coder.encode(["address"], ["0xdd99b75f095d0c4d5112aCe938e4e6ed962fb024"]);
+  const bytecode = hardhatEthers.utils.solidityPack(
+    ["bytes32", "bytes", "bytes"],
+    [saltHash, ExtensionRegistry.bytecode.object, args],
+  );
+  const bytecodeWithoutSalt = hardhatEthers.utils.solidityPack(
+    ["bytes", "bytes"],
+    [ExtensionRegistry.bytecode.object, args],
+  );
+
+  const deployInfoPacked = hardhatEthers.utils.solidityPack(
+    ["bytes1", "address", "bytes32", "bytes32"],
+    ["0xff", commonFactory, saltHash, hardhatEthers.utils.solidityKeccak256(["bytes"], [bytecodeWithoutSalt])],
+  );
+  const predictedAddress = hardhatEthers.utils.solidityKeccak256(["bytes"], [deployInfoPacked]);
+
+  // console.log(bytecode);
+  // console.log(bytecode == savedBytecode);
+
+  // create deployment tx
+  let tx = {};
+  const block = await hardhatEthers.provider.getBlock("latest");
+  const baseBlockFee = block && block.baseFeePerGas ? block.baseFeePerGas : hardhatEthers.utils.parseUnits("1", "gwei");
+  const feeData = await hardhatEthers.provider.getFeeData();
+  console.log(feeData);
+
+  if (feeData.maxFeePerGas && feeData.maxPriorityFeePerGas) {
+    const baseMaxFeePerGas = baseBlockFee.mul(2);
+    const maxPriorityFeePerGas = BigNumber.from(feeData.maxPriorityFeePerGas);
+    tx = {
+      from: signer.address,
+      to: commonFactory,
+      value: 0,
+      nonce: await signer.getTransactionCount("latest"),
```

Comment:
@jakeloo what nonce are you supposed to pass in the tx when you want to craft a keyless tx? the nonce of the "crafted" tx signer? seems backwards since you might need it to craft the tx itself

Reply:
found it - you always hardcode to 0 and assume the derived EOA address has never been used before :p